### PR TITLE
Add SentenceStatus field and create generic UNKNOWN_REMOVED_FROM_SOURCE enum value

### PR DESCRIPTION
**Pull request by terinpw**

### CODE REVIEW

File: recidiviz/common/constants/enum_canonical_strings.py [hunk 1]
```diff
@@ -37,6 +37,7 @@
 # default value for enums that are not provided (which should be represented
 # with None/NULL).
 external_unknown = 'EXTERNAL_UNKNOWN'
+unknown_removed_from_source = 'UNKNOWN_REMOVED_FROM_SOURCE'
```
Comment (zdg2102) [inline recidiviz/common/constants/enum_canonical_strings.py]: Might be worth adding a comment clarifying what this is, like `external_unknown` has

File: recidiviz/common/constants/enum_canonical_strings.py [hunk 2]
```diff
@@ -105,6 +106,12 @@
 bond_status_active = 'ACTIVE'
 bond_status_posted = 'POSTED'
 
+# Sentence
+
+sentence_status_commuted = 'COMMUTED'
+sentence_status_completed = 'COMPLETED'
```
Comment (zdg2102) [inline recidiviz/common/constants/enum_canonical_strings.py]: Probably shouldn't address this in this PR, but should we remove `SENTENCE_COMPLETED` from `ChargeStatus` and instead create a dummy sentence with `SentenceStatus.COMPLETED`?

File: recidiviz/common/constants/enum_canonical_strings.py (hunk 1 above)
Comment (terinpw) [reply recidiviz/common/constants/enum_canonical_strings.py]: Added!

File: recidiviz/common/constants/enum_canonical_strings.py (hunk 2 above)
Comment (terinpw) [reply recidiviz/common/constants/enum_canonical_strings.py]: Filed issue #663

**Comment by terinpw** [comment]
helps with #585

**Pull request merged**